firmware/shared/src/app/app_utils.h

```diff
@@ -6,8 +6,14 @@
 /* @brief Extract the basename from a file path */
 #define __BASENAME__(path) (__builtin_strrchr(path, '/') ? __builtin_strrchr(path, '/') + 1 : path)
 
-#define MIN(x, y) (((x) < (y)) ? (x) : (y))
-#define MAX(x, y) (((x) > (y)) ? (x) : (y))
+#ifndef MIN
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+#endif
+
+#ifndef MAX
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+#endif
```

Comment:
did something happen?

Reply:
STM defined it's own min and max macros :/. Painful necessity.